snafu-derive/src/lib.rs

```diff
@@ -347,12 +627,15 @@ fn parse_snafu_struct(
         return Err(vec![one_field_error(span)]);
     }
 
-    let transformation = match transformation {
-        Some((ty, expr)) => Transformation::Transform { ty, expr },
-        None => Transformation::None {
+    let (maybe_transformation, errs) = transformations.finish();
+    let transformation = maybe_transformation
+        .map(|(ty, expr)| Transformation::Transform { ty, expr })
+        .unwrap_or(Transformation::None {
             ty: inner.into_value().ty,
-        },
-    };
+        });
```

Comment:
FYI, [`unwrap_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_else) and [`map_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or_else) exist.

This currently unconditionally calls `into_value`, which is a *slight* inefficiency. I'm not really worried about it here or in the earlier case of it.

Reply:
Oops; that line was a copy/paste, and I missed the change in behavior in this new context.  Will fix shortly.